source/guide_salt.rst

```diff
@@ -0,0 +1,132 @@
+.. author:: ctr49 <https://github.com/ctr49>
+
+.. tag:: automation
+
+.. highlight:: console
+
+.. sidebar:: Logo
+
+  .. image:: _static/images/saltstack.png
+      :align: center
+
+##########
+SaltStack
+##########
+
+.. tag_list::
+
+`SaltStack`_ (or simply `Salt`) is Python-based, open-source software for event-driven IT automation, remote task execution, and configuration management. Supporting the "Infrastructure as Code" approach to data center system and network deployment and management, configuration automation, SecOps orchestration, vulnerability remediation, and hybrid cloud control.
+
+The ``salt-master`` controls so called ``salt-minions`` through formulae, pillars and grains. The scope of this guide is to install a ``salt-master``(without a co-located minion) on an Uberspace.
+
+----
+
+.. note:: For this guide you should be familiar with the basic concepts of
+
+  * :manual:`Opening ports <basics-ports>`
+  * :manual:`supervisord <daemons-supervisord>`
+
+Prerequisites
+=============
+
+We need two open TCP ports for minions to communicate with the Salt master.
+
+::
+
+ [isabell@stardust ~]$ uberspace port add
+ Port <high-port1> will be open for TCP and UDP traffic in a few minutes.
+ [isabell@stardust ~]$ uberspace port add
+ Port <high-port2> will be open for TCP and UDP traffic in a few minutes.
+ [isabell@stardust ~]$ uberspace port list
+ <high-port1>
+ <high-port2>
+
+Take note of the high ports, you will need them for salt master configuration later.
+
+Installation
+============
+
+Install salt-master
+-------------------
+
+The ``salt`` installation script requires root privileges and only supports virtualenv installation for Ubuntu. So we'll follow the "Developer installation" guide to install the salt master manually. However, we don't want the latest head development version, but the latest stable/tagged version. As of writing this guide, the latest stable version is 2019.2.0
+
+::
+
+ [isabell@stardust ~]$ mkdir -p ~/salt/src
+ [isabell@stardust ~]$ cd ~/salt/src
+ [isabell@stardust ~]$ git clone https://github.com/saltstack/salt
+ [isabell@stardust ~]$ cd salt
+ [isabell@stardust ~]$ git checkout tags/v2019.2.0
+ [isabell@stardust ~]$ virtualenv ~/salt/virtualenv
+ [isabell@stardust ~]$ source ~/salt/virtualenv/bin/activate
+ [isabell@stardust ~]$ pip install pyzmq PyYAML pycrypto msgpack-python jinja2 psutil futures tornado
+ [isabell@stardust ~]$ MIMIC_SALT_INSTALL=1 pip install --global-option='--salt-root-dir=~/salt/virtualenv/' -e ~/salt/src/salt
```

Comment:
This command fails for me :/

```
Installing collected packages: msgpack, six, singledispatch, backports-abc, tornado, salt
  Running setup.py install for msgpack ... error
    Complete output from command /usr/bin/python2 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-lOHZtg/msgpack/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" --salt-root-dir=~/salt/virtualenv/ install --record /tmp/pip-39_Eop-record/install-record.txt --single-version-externally-managed --compile:
    /usr/lib64/python2.7/distutils/dist.py:267: UserWarning: Unknown distribution option: 'long_description_content_type'
      warnings.warn(msg)
    /usr/lib64/python2.7/distutils/dist.py:267: UserWarning: Unknown distribution option: 'project_urls'
      warnings.warn(msg)
    usage: -c [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
       or: -c --help [cmd1 cmd2 ...]
       or: -c --help-commands
       or: -c cmd --help
    
    error: option --salt-root-dir not recognized
    
    ----------------------------------------
Command "/usr/bin/python2 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-lOHZtg/msgpack/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" --salt-root-dir=~/salt/virtualenv/ install --record /tmp/pip-39_Eop-record/install-record.txt --single-version-externally-managed --compile" failed with error code 1 in /tmp/pip-build-lOHZtg/msgpack/
```

Reply:
I may have a newer pip version in my virtualenv, can you please try upgrading it?
pip install --upgrade pip
I will adjust the guide if that is really needed.

Looking at your output I'm just realizing, that python27 is being used as it still seems to be the default python. What do you think about adapting this to python3?

BR,
   Christian

Reply:
> What do you think about adapting this to python3?

Sure, use whatever software version works best for salt.

Reply:
using pip-19.1.1 and py3.6 also doesn't fix this for me :/